frontend/src/scenes/dashboard/DashboardItems.js

```diff
@@ -0,0 +1,166 @@
+                                                            onClick={() =>
+                                                                setColors({ ...colors, [item.id]: className })
+                                                            }
+                                                        >
+                                                            <span
+                                                                style={{
+                                                                    background: allColorStyles[className],
+                                                                    border: '1px solid #eee',
+                                                                    display: 'inline-block',
+                                                                    width: 13,
+                                                                    height: 13,
+                                                                    verticalAlign: 'middle',
+                                                                    marginRight: 5,
+                                                                    marginBottom: 1,
+                                                                }}
+                                                            />
+                                                            {color}
+                                                        </Menu.Item>
+                                                    ))}
+                                                </Menu>
+                                            }
+                                        >
+                                            <span style={{ cursor: 'pointer', marginTop: -3 }}>
+                                                <BgColorsOutlined />
+                                            </span>
+                                        </Dropdown>
+                                    )}
+                                    <Dropdown
+                                        placement="bottomRight"
+                                        trigger="click"
+                                        overlay={
+                                            <Menu>
+                                                <Menu.Item icon={<Icon />} onClick={() => router.actions.push(link)}>
+                                                    {viewText}
+                                                </Menu.Item>
+                                                <Menu.Item
+                                                    icon={<EditOutlined />}
+                                                    onClick={() => renameDashboardItem(item.id)}
+                                                >
+                                                    Rename
+                                                </Menu.Item>
+                                                <Menu.Item icon={<DeleteOutlined />} className="text-danger">
```

Comment:
For some reason, if you refresh this page, the first time you click it it doesn't work. Only on the second try

Reply:
it was actually that you had to directly hit the text "delete", clicking the icon before it or the space after it caused it to not trigger the right event